src/com/cdd/bao/editor/SearchSchema.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * BioAssay Ontology Annotator Tools
+ * 
+ * (c) 2014-2017 Collaborative Drug Discovery Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License 2.0
+ * as published by the Free Software Foundation:
+ * 
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.cdd.bao.editor;
+
+import java.util.*;
+
+import com.cdd.bao.editor.EditSchema.Branch;
+import com.cdd.bao.template.Schema;
+
+import javafx.scene.Node;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeView;
+
+public class SearchSchema
+{
+	// Houses search state to facilitate stepping through results.
+	public static class State
+	{
+		String searchText;
+		int index;
+		List<TreeItem<Branch>> found;
+	}
+
+	// return true if search text matches any of uri, name, or description.
+	private static boolean isMatch(String uri, String name, String descr, String searchText)
+	{
+		boolean isMatch = false;
+		if ((uri != null && uri.indexOf(searchText) >= 0)
+			|| (name != null && name.indexOf(searchText) >= 0)
+			|| (descr != null && descr.indexOf(searchText) >= 0))
+		{
+			isMatch = true;	
+		}
+		return isMatch;
+	}
+
+	// return list of nodes that match search text.
+	public static List<TreeItem<Branch>> find(TreeView<Branch> treeView, String searchText)
+	{
+		List<TreeItem<Branch>> found = new ArrayList<>();
+		Deque<TreeItem<Branch>> stack = new ArrayDeque<>();
+
+		// Maybe we should skip over the root since it is not shown.
+		stack.addFirst(treeView.getRoot());
+		while (stack.size() > 0)
+		{
+			TreeItem<Branch> curItem = stack.removeFirst();
+			Branch curBranch = curItem.getValue();
+			if (curBranch.group != null)
+			{
+				Schema.Group sg = curBranch.group;
+				if (isMatch(sg.groupURI, sg.name, sg.descr, searchText))
```

Comment:
You can use more columns and stick the payload of the if-statement on the same line; there's no hard rule for this, but brevity is preferred if there's no good counterargument.